apps/ejabberd/src/mod_version.erl

```diff
@@ -26,40 +26,40 @@ process_iq(From, _To, #iq{type = get} = IQ) ->
     Host = From#jid.lserver,
     IQ#iq{type = result,
           sub_el =
-              [#xmlel{name = <<"query">>,
-                      attrs = [{<<"xmlns">>, ?NS_VERSION}],
-                      children =
-                          [#xmlel{name = <<"name">>, attrs = [],
-                                  children =[#xmlcdata{content = Name}]},
-                           #xmlel{name = <<"version">>, attrs = [],
-                                  children =[#xmlcdata{content = Version}]}
-			  ] ++ add_os_info(Host)}]}.
+          [#xmlel{name = <<"query">>,
+                  attrs = [{<<"xmlns">>, ?NS_VERSION}],
+                  children =
+                  [#xmlel{name = <<"name">>, attrs = [],
+                          children =[#xmlcdata{content = Name}]},
+                   #xmlel{name = <<"version">>, attrs = [],
+                          children =[#xmlcdata{content = Version}]}
+                  ] ++ add_os_info(Host)}]}.
 
--spec add_os_info(binary()) -> [#xmlel{}] | [].
+-spec add_os_info(binary()) -> [exml:element()] | [].
 add_os_info(Host) ->
     case gen_mod:get_module_opt(Host, ?MODULE, os_info, false) of
-	true ->
-	    [#xmlel{name = <<"os">>, attrs = [],
-		    children = [#xmlcdata{content = os_info()}]}];
-	_ ->
-	    []
-    end.   
+        true ->
+            [#xmlel{name = <<"os">>, attrs = [],
+                    children = [#xmlcdata{content = os_info()}]}];
+        _ ->
+            []
+    end.
 
 -spec mongoose_info() -> {binary(), binary()}.
 mongoose_info() ->
-    {ok, Version} = application:get_key(mongooseim, vsn),
-    {ok, Name} = application:get_key(mongooseim, description),
-    {list_to_binary(Name), list_to_binary(Version)}.
+    {ok, EjdMimVsn} = application:get_key(ejabberd, vsn), %returns 2.1.8+mim-*
+    Version = string:sub_string(EjdMimVsn, 11),
```

Comment:
Just a small comment, but I find `"2.1.8+mim-" ++ Version = EjdMimVsn` (or `<<"2.1.8+mim-", Version/binary>> = list_to_binary(EjdMimVsn)` as binary pattern matching is more common than string pattern matching and we need a binary Version anyway) more readable in that it doesn't require a comment and a magic `11`.